imap_processing/tests/lo/test_lo_science.py

```diff
@@ -266,3 +300,51 @@ def test_validate_parse_events(sample_data, attr_mgr):
 
     assert dataset["de_count"].values == 1998
     assert dataset["passes"].values == 8
+
+
+def test_organize_spin_data(fake_spin_data):
+    # Arrange
+    expected_dataset = xr.Dataset(
+        data_vars=dict(
+            num_completed=(["epoch"], np.array([0, 0])),
+            acq_start_sec=(["epoch"], np.array([0, 0])),
+            acq_start_subsec=(["epoch"], np.array([0, 0])),
+            acq_end_sec=(["epoch"], np.array([0, 0])),
+            acq_end_subsec=(["epoch"], np.array([0, 0])),
+            start_sec_spin=(
+                ["epoch", "spin"],
+                np.array([np.arange(0, 56, 2), np.arange(1, 57, 2)]),
+            ),
+            start_subsec_spin=(
+                ["epoch", "spin"],
+                np.array([np.arange(0, 56, 2), np.arange(1, 57, 2)]),
+            ),
+            esa_neg_dac_spin=(
+                ["epoch", "spin"],
+                np.array([np.arange(0, 56, 2), np.arange(1, 57, 2)]),
+            ),
+            esa_pos_dac_spin=(
+                ["epoch", "spin"],
+                np.array([np.arange(0, 56, 2), np.arange(1, 57, 2)]),
+            ),
+            valid_period_spin=(
+                ["epoch", "spin"],
+                np.array([np.arange(0, 56, 2), np.arange(1, 57, 2)]),
+            ),
+            valid_phase_spin=(
+                ["epoch", "spin"],
+                np.array([np.arange(0, 56, 2), np.arange(1, 57, 2)]),
+            ),
+            period_source_spin=(
+                ["epoch", "spin"],
+                np.array([np.arange(0, 56, 2), np.arange(1, 57, 2)]),
+            ),
+        ),
+        coords=dict(epoch=(["epoch"], np.array([0, 1]))),
```

Comment:
Should spin also be a coordinate? Or we should add it possibly?
```suggestion
        coords=dict(epoch=(["epoch"], np.array([0, 1])), spin=(["spin"], np.arange(28))),
```

Reply:
I add that as a coordinate in the CDF attribute part of the code, so I have that being done on another branch which will be my next PR